src/components/media-views.js

```diff
@@ -837,3 +846,150 @@ AFRAME.registerComponent("media-image", {
     this.el.emit("image-loaded", { src: this.data.src, projection: projection });
   }
 });
+
+AFRAME.registerComponent("media-pdf", {
+  schema: {
+    src: { type: "string" },
+    projection: { type: "string", default: "flat" },
+    contentType: { type: "string" },
+    index: { default: 0 },
+    batch: { default: false }
+  },
+
+  init() {},
+
+  remove() {
+    if (this.data.batch && this.mesh) {
+      this.el.sceneEl.systems["hubs-systems"].batchManagerSystem.removeObject(this.mesh);
+    }
+
+    if (this.currentPageTextureIsRetained) {
+      const cacheKey = this.currentTextureCacheKey();
+      textureCache.release(cacheKey);
+      this.currentPageTextureIsRetained = false;
+    }
+  },
+
+  currentTextureCacheKey() {
+    // We ensure a unique texture for each PDF entity, because they are drawn over during pagination.
+    return `${this.el.object3D.uuid}_${this.canvas.width}_${this.canvas.height}`;
```

Comment:
The main point of the texture cache is to re-use textures across multiple object3Ds. If we key it light this it would probably be better just to have a single `CanvasTexture` per `media-pdf` and just resize its canvas as needed, avoiding the texture cache al together. 

If we wanted to actually share textures across multiple instances of the same PDF, its possible, but you would probably end up needing one texture per page, which seems probably worse overall, just to handle the uncommon case of displaying multiple instances of the same page of a PDF.

Reply:
yep that makes sense. earlier versions of this code was trying to cache at the page level and now that I landed on a texture-per-entity the cache makes no sense.